apps/www/registry/default/ui/phone-input.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react"
+import { Check, ChevronsUpDown } from "lucide-react"
+import * as RPNInput from "react-phone-number-input"
+import flags from "react-phone-number-input/flags"
+
+import { cn } from "@/lib/utils"
+import { Button } from "@/registry/default/ui/button"
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from "@/registry/default/ui/command"
+import { Input, InputProps } from "@/registry/default/ui/input"
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/registry/default/ui/popover"
+
+export type PhoneInputValue = RPNInput.Value
+
+type PhoneInputProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  "onChange"
+> &
+  RPNInput.Props<typeof RPNInput.default>
+
+const PhoneInput: React.ForwardRefExoticComponent<PhoneInputProps> =
+  React.forwardRef<React.ElementRef<typeof RPNInput.default>, PhoneInputProps>(
+    ({ className, onChange, ...props }, ref) => (
+      <RPNInput.default
+        ref={ref}
+        placeholder={"Enter a phone number"}
```

Comment:
What do you think about letting the placeholder come from the props? Or, because it is a specific component, you could do something like that:

``` typescript
placeholder={props.placeholder ?? "Enter a phone number"}
```

Reply:
I removed the default placeholder. Assignment to the prop is already optional.

This seems to be the cleanest way. 👌